### ISSUE TITLE
Check-in/out issue/error

### ISSUE THREAD

**Issue by janequist** [inveniosoftware/react-invenio-app-ils#269]
Package version (if known): alpha.14

## Describe the bug
Trying to understand the process to get Extension on Loan during which encountered a multi-part error given below:

## Steps to Reproduce
1. The Document required was on loan (to patron3) - so first as admin, checked in loan for 
[PROCEEDING Consectetur ipsum porro magnam dolore dolor.]
Check-in was successful

2. Then as user 'admin' took out Loan for same and Checked out the physical copy that was available(CAN_CIRCULATE) - Check-out was successful

3. Tried to view the same on Frontsite as same user 'admin'
Issue 1.a - it gave me option to request the same item on loan again (i thought it should give me an option to Extend)
Error 1.b - when made the request, it gave an error saying patron3 has an active loan for same (i Checked it out as admin and patron3 loan was Checked-in before that. So it seems the previous check-in did not fully work - it worked partially in that it allowed me to Check-out).

4. Error 2. Then as same user 'admin' tried to look at "Your Loans" on Frontsite dropdown menu
App exited - got below error on browser:
×
TypeError: loan.metadata.end_date.diffNow is not a function
LoanExtendButton.validate
src/lib/pages/frontsite/PatronProfile/PatronCurrentLoans/LoanExtendButton.js:70
  67 | if (!hasExtendAction)
  68 |   return { isValid: false, msg: INFO_MESSAGES.MISSING_ACTION_URL };
  69 | const { loanWillExpireDays } = invenioConfig.CIRCULATION;
> 70 | const isTooEarly =
     | ^  71 |   loan.metadata.end_date.diffNow('days').days > loanWillExpireDays;
  72 | if (isTooEarly) {
  73 |   return {
View compiled
LoanExtendButton.render
src/lib/pages/frontsite/PatronProfile/PatronCurrentLoans/LoanExtendButton.js:94
  91 | render() {
  92 |   const { loan } = this.props;
  93 |   const { isLoading } = this.state;
> 94 |   const validation = this.validate(loan);
     | ^  95 |   const isDisabled = !validation.isValid;
  96 | 
  97 |   return (
View compiled
▶ 22 stack frames were collapsed.
(anonymous function)
src/lib/modules/Patron/PatronCurrentLoans/actions.js:30
  27 | });
  28 | try {
  29 |   const response = await loanApi.list(selectQuery(patronPid, page, size));
> 30 |   dispatch({
     | ^  31 |     type: SUCCESS,
  32 |     payload: response.data,
  33 |   });

**Comment by janequist** [comment]
Update: When I restarted and checked "Your Loans", the item was in my Current loan list and I am able to request Extension. So it seems the previous Check-out was successful and the consequent errors dont seem to be user/usage error.

**Comment by FlorianCassayre** [comment]
Thanks for testing the app and reporting the error.
It looks like to me that you are using an older version of `react-invenio-app-ils`, and I believe the problem you encountered was fixed in [`6d00569`](https://github.com/inveniosoftware/react-invenio-app-ils/commit/6d00569f22be169d4d9f01ab6b11e031302893ec#diff-c3b19ee679d745cba0c0af3e7d9a76d2da00ab68e1dac1887a44ed2de519a404R73). We refactored the frontend serialization logic and in consequence it (temporarily) broke some components, such as this one.
